src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -290,7 +290,12 @@ public override FieldLayoutAlgorithm GetLayoutAlgorithmForType(DefType type)
             else if (_simdHelper.IsVectorOfT(type))
                 return _vectorOfTFieldLayoutAlgorithm;
             else
-                return _metadataFieldLayoutAlgorithm;
+                return GetMetadataFieldLayoutAlgorithm();
+        }
+
+        protected virtual FieldLayoutAlgorithm GetMetadataFieldLayoutAlgorithm()
```

Comment:
I don't think this factoring is going to work. We'll also need the other algorithms here to be special eventually. I think you should just override `GetLayoutAlgorithmForType`, copy the implementation from here with a tweak, and be done with it.

Reply:
OK, will do.